woocommerce/woocommerce-ios
# Product Selector: Filter search results

Part of #6489 


### Description

Since we have filters on the product selector screen, it is important that product searches take into account filters to return the correct filters.

This PR updates all layers of the app to support searching for products with filters:
- Networking: Add optional parameters for filters when searching products. 
- Yosemite: Update ProductAction and ProductStore with filters.
- UI: Send filters to `searchProducts` action

### Testing instructions

- Open the product selector screen from the Edit Coupon flow.
- Select Filter, then select a category that filters out some of the products.
- Close the product selector and navigate back to the Products tab, notice that the tab still shows the full list of products.
- Repeat the first 2 steps, then type a keyword in the search bar to find a product that doesn't exist with the filters applied. The result should be empty.
- Keep the search query, and clear the filters. The matching item should show up in the results.

### Screenshots


https://user-images.githubusercontent.com/5533851/165721820-9d202529-783d-4ec6-bd58-56a72d45408e.mp4



---
